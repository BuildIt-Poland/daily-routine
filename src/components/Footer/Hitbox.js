import styled from 'styled-components';

import { fontMedium, colorOpaqueBlack, colorPink, fontWeightNormal, borderRadius } from '../../styles/designTokens';

const Hitbox = styled.a`
  display: flex;
  height: 3.6rem;
  white-space: nowrap;
  align-items: center;
  text-decoration: none;
  font-size: ${fontMedium};
  font-weight: ${fontWeightNormal};
  color: ${colorOpaqueBlack};
  border-radius: ${borderRadius};
  outline: 0;
  transition: all 150ms linear;

  &.focus-visible {
    box-shadow: -0.4rem 0 0 0 ${colorPink}, 0.4rem 0 0 0 ${colorPink};
    background-color: ${colorPink};
  }
`;

export default Hitbox;
