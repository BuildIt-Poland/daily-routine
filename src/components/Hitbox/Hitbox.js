// Button wrapper with all styles removed
// Useful for accessible action icons / target areas

import styled from 'styled-components';

const Hitbox = styled.button`
  position: relative;
  border: 0;
  margin: 0;
  padding: 0;
  background: none;
  display: flex;
  align-items: center;
  justify-content: center;
  min-width: 4rem;
  min-height: 4rem;
  cursor: pointer;
  outline: 0;
`;

export default Hitbox;
