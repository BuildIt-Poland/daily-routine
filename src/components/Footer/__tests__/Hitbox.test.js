import React from 'react';
import { create } from 'react-test-renderer';

import Hitbox from '../Hitbox';

describe('COMPONENT - Footer Hitbox', () => {
  it('renders correctly', () => {
    const component = create(<Hitbox />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
