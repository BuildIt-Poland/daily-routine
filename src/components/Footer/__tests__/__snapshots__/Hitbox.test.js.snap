// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Footer Hitbox renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 3.6rem;
  white-space: nowrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.4rem;
  font-weight: 400;
  color: rgba(34,34,34,0.4);
  border-radius: 0.6rem;
  outline: 0;
  -webkit-transition: all 150ms linear;
  transition: all 150ms linear;
}

.c0.focus-visible {
  box-shadow: -0.4rem 0 0 0 #ffdcfa,0.4rem 0 0 0 #ffdcfa;
  background-color: #ffdcfa;
}

<a
  className="c0"
/>
`;
